fault-proof/tests/e2e.rs

```diff
@@ -293,20 +287,178 @@ mod e2e {
+            &proposer_signer,
+            &l1_rpc_url,
+            env.deployed.factory,
+            Bytes::from(set_impl_call.abi_encode()),
+            None,
+        )
+        .await?;
+
+        let legacy_game_type_call = MockOptimismPortal2::setRespectedGameTypeCall {
+            _gameType: MOCK_PERMISSIONED_GAME_TYPE,
+        };
+        send_contract_transaction(
+            &proposer_signer,
+            &l1_rpc_url,
+            env.deployed.portal,
+            Bytes::from(legacy_game_type_call.abi_encode()),
+            None,
+        )
+        .await?;
+
+        let mut expected_index = initial_game_count;
+        info!("Seeding legacy game (type {MOCK_PERMISSIONED_GAME_TYPE})");
+
+        let proposer_handle = start_proposer(
+            &env.rpc_config,
+            PROPOSER_PRIVATE_KEY,
+            &env.deployed.factory,
+            TEST_GAME_TYPE,
+        )
+        .await?;
+        info!("✓ Proposer started after legacy games seeded");
+
+        let l2_provider = ProviderBuilder::default().connect_http(env.rpc_config.l2_rpc.clone());
+        let interval = 10;
+        let l2_block = U256::from(env.anvil.starting_l2_block_number + interval);
+        let root_claim = l2_provider.compute_output_root_at_block(l2_block).await?;
+        let extra_data = <(U256, u32)>::abi_encode_packed(&(l2_block, u32::MAX));
+
+        let create_call = DisputeGameFactory::createCall {
+            _gameType: MOCK_PERMISSIONED_GAME_TYPE,
+            _rootClaim: root_claim,
+            _extraData: Bytes::from(extra_data.clone()),
+        };
+
+        send_contract_transaction(
+            &proposer_signer,
+            &l1_rpc_url,
+            env.deployed.factory,
+            Bytes::from(create_call.abi_encode()),
+            Some(init_bond),
+        )
+        .await?;
+
+        // Wait for the game to be indexed by the factory
+        loop {
+            let current = factory_reader.gameCount().call().await?;
+            if current > expected_index {
+                expected_index = current;
+                break;
+            }
+            sleep(Duration::from_millis(200)).await;
         }
 
+        let game_info = factory_reader.gameAtIndex(expected_index - U256::from(1)).call().await?;
+        let mock_game_address = game_info.proxy_;
+        assert_eq!(game_info.gameType_, MOCK_PERMISSIONED_GAME_TYPE);
+        info!(" • Mock permissioned game at {mock_game_address}");
+
+        let restore_type_call =
+            MockOptimismPortal2::setRespectedGameTypeCall { _gameType: TEST_GAME_TYPE };
+        send_contract_transaction(
+            &proposer_signer,
+            &l1_rpc_url,
+            env.deployed.portal,
+            Bytes::from(restore_type_call.abi_encode()),
+            None,
+        )
+        .await?;
+        info!("✓ Respected game type restored to {TEST_GAME_TYPE}");
+
+        let factory = DisputeGameFactory::new(env.deployed.factory, env.anvil.provider.clone());
```

Comment:
```
        let factory_reader =
            DisputeGameFactory::new(env.deployed.factory, env.anvil.provider.clone());
```
nit: I think we can reuse the factory initialized above.

Reply:
[4e4b972](https://github.com/succinctlabs/op-succinct/pull/677/commits/4e4b972b1ab03c117f95b78726531a24482d77e7)